build.assets/README.md

```diff
@@ -9,7 +9,18 @@ It is a part of Gravitational CI/CD pipeline. To build Teleport type:
 make
 ```
 
-### DynamoDB static binary docker build 
+# Safely updating build box Dockerfiles
+
+The build box images are used in Drone pipelines and GitHub Actions and the resulting image is
+pushed to Amazon ECR. This means that to safely introduce changes to Dockerfiles, those changes
+should be split into two stages:
+
+1. First you open a PR which updates a Dockerfile and get the PR merged.
+2. Once it's merged, Drone is going to pick it up, build a new build box image and push it to Amazon
+   ECR.
```

Comment:
One thing I don't fully grok is why not introducing the changes in two phases was a problem for Drone but not for GHA.

Is it because of this?

https://github.com/gravitational/teleport/blob/4a8658ab250f29fded740206c63eaabe5ba0210f/build.assets/Makefile#L435-L450

Reply:
I don't fully understand either. Looking at [previous logs](https://pipelines.actions.githubusercontent.com/serviceHosts/873ff56f-3863-4e57-be35-8da0b929ee9e/_apis/pipelines/1/runs/206691/signedlogcontent/2?urlExpires=2023-01-23T17%3A26%3A05.1309529Z&urlSigningMethod=HMACV1&urlSignature=uCn6AMk%2FKcaYCVHeDc4JZaM7n%2FA42e0lzdY5zfl0oog%3D) it seems like it pulls a builder from ghcr.io, so I would expect the same caveats from Drone to apply.

Reply:
> One thing I don't fully grok is why not introducing the changes in two phases was a problem for Drone but not for GHA.

I think it was because with drone we shared a single repo for all buildbox images and tags, whereas with GHA, we *maybe* have some per-job isolation layer

The problems we were seeing were:  E.g. if a old build was run on a branch, it would re-push a tag over the newer buildbox, which would then cause newer builds to fail when they pulled the out of date buildbox.  The fix for this was to prevent jobs from updating the buildbox for a branch unless separately triggered.

During these troubles, we had some good discussions on motivations and dreams for buildbox determinism and reproducibility:

* https://github.com/gravitational/teleport/pull/10611
* https://github.com/gravitational/teleport/pull/10206

I personally lost the thread after switching teams and going on leave for a bit, but I'm interested to catch up on what is going on.